aosp_diff/base_aaos/packages/modules/Bluetooth/0001-Fix-metadata-issue-when-reconnect-pixel.patch

```diff
@@ -1,52 +1,192 @@
+     }
+ }
++
++flag {
++    name: "associate_browse_l2cap_request_with_active_control_channel"
++    namespace: "bluetooth"
++    description: " Associate a browse l2cap channel connection with the control channel from that same device regardless of specified role"
++    bug: "373482555"
++    metadata {
++        purpose: PURPOSE_BUGFIX
++    }
++}
+\ No newline at end of file
 diff --git a/system/bta/av/bta_av_act.cc b/system/bta/av/bta_av_act.cc
-index e94c26d51f..36aab0f070 100644
+index 36aab0f070..725f8e6062 100644
 --- a/system/bta/av/bta_av_act.cc
 +++ b/system/bta/av/bta_av_act.cc
-@@ -664,8 +664,13 @@ void bta_av_rc_opened(tBTA_AV_CB* p_cb, tBTA_AV_DATA* p_data) {
+@@ -664,13 +664,13 @@ void bta_av_rc_opened(tBTA_AV_CB* p_cb, tBTA_AV_DATA* p_data) {
    if ((p_cb->features & BTA_AV_FEAT_BROWSE) &&
        (rc_open.peer_features & BTA_AV_FEAT_BROWSE) &&
        ((p_cb->rcb[i].status & BTA_AV_RC_ROLE_MASK) == BTA_AV_RC_ROLE_INT)) {
--    log::verbose("opening AVRC Browse channel");
--    AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
-+    log::verbose("Not opening AVRC Browse channel as INT");
-+    /* Commenting the below line as for IVI use case, for A2DP SNK, we dont
-+     * support browsing channel. So no need to open browsing channel as
-+     * INITIATOR. Revert this when have logic to resolve this conflict in
-+     * stack/avct.
-+     */
-+    // AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
+-    log::verbose("Not opening AVRC Browse channel as INT");
++    log::verbose("abhadour Not opening AVRC Browse channel as INT");
+     /* Commenting the below line as for IVI use case, for A2DP SNK, we dont
+      * support browsing channel. So no need to open browsing channel as
+      * INITIATOR. Revert this when have logic to resolve this conflict in
+      * stack/avct.
+      */
+-    // AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
++      AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
    }
  }
  
+diff --git a/system/stack/avct/avct_bcb_act.cc b/system/stack/avct/avct_bcb_act.cc
+index c4ad431904..1d73c45bac 100644
+--- a/system/stack/avct/avct_bcb_act.cc
++++ b/system/stack/avct/avct_bcb_act.cc
+@@ -28,6 +28,7 @@
+ #define LOG_TAG "bluetooth"
+ 
+ #include <bluetooth/log.h>
++#include <com_android_bluetooth_flags.h>
+ #include <string.h>
+ 
+ #include "avct_api.h"
+@@ -38,6 +39,8 @@
+ #include "osi/include/allocator.h"
+ #include "osi/include/osi.h"
+ #include "stack/avct/avct_defs.h"
++#include "stack/avct/avct_int.h"
++#include "stack/include/avct_api.h"
+ #include "stack/include/bt_hdr.h"
+ #include "stack/include/bt_types.h"
+ 
+@@ -155,57 +158,69 @@ void avct_bcb_unbind_disc(tAVCT_BCB* /* p_bcb */, tAVCT_LCB_EVT* p_data) {
+  * Returns          Nothing.
+  *
+  ******************************************************************************/
++namespace {
++bool is_valid_role_check(const tAVCT_CCB* p_ccb) {
++  log::warn("abhadour Inside valid role check");
```

Comment:
Logs should be clean in all places.

Reply:
PR for custom EB, will clean up after confirmation